byte_embeddings.py

```diff
@@ -2,95 +2,77 @@
-    n_bytes = 2
-    window = hex_chars * n_bytes
-    byte_word_array = []
-    for idx in range(0, hex_len, window):
-        byte_word = hex_string[idx: idx+window].decode('utf-8')
-        byte_word_array.append(byte_word)
-    return byte_word_array
-
-
-def train_model(b_words, model):
+    """
     if model:
         print("Updating model with {} files".format(len(b_words)))
         model.build_vocab(b_words, update=True)
     else:
         print("Training new model")
-        model = Word2Vec(size=256, window=2, workers=8, sg=1, hs=0, negative=5)
+        model = Word2Vec(size=256, window=2,
+                         workers=8, sg=1, hs=0,
+                         negative=5, min_count=0)
         model.build_vocab(b_words)
-    model.train(b_words, total_examples=len(b_words), epochs=15, compute_loss=True)
+    model.train(b_words, total_examples=len(b_words),
+                epochs=15, compute_loss=True)
     return model
 
-def generate_batch(zip_file, file_names, cur_idx, batch_size, password=None):
-    batch = []
-    print("Loading Batch")
-    n_byte_words = 0
-    for f in file_names[cur_idx:(cur_idx+batch_size)]:
-        b_string = load_file(zip_file, f, password)
-        b_words = split_string(b_string)
-        batch.append(b_words)
-        n_byte_words += len(b_words)
-    print("Average number of byte words:", n_byte_words/float(batch_size))
-    return batch
-
 
 if __name__ == '__main__':
     parser = argparse.ArgumentParser(description='Extract zip files')
 
-    parser.add_argument('--zip_file_name', help='name of zip containing malware')
+    parser.add_argument('--zip_file_name',
+                        help='name of zip containing malware')
     parser.add_argument('--pass_string', help='password')
-    parser.add_argument('--batch_size', default=4, 
+    parser.add_argument('--batch_size', default=4,
+                        help='number of binaries to add to model at a time')
+    parser.add_argument('--existing_model',
+                        default=None,
                         help='number of binaries to add to model at a time')
-    parser.add_argument('--existing_model', default=None, help='number of binaries to add to model at a time')
     args = parser.parse_args()
 
-    zip_file = ZipFile(args.zip_file_name, 'r')
-
-    fn = get_file_names(zip_file)
+    zdh = ZipDataHandler(args.zip_file_name, password=args.pass_string)
 
-    random.shuffle(fn)
+    zdh.shuffle_file_names()
 
     if args.existing_model:
         w2v_model = Word2Vec.load(args.existing_model)
     else:
-        w2v_model=None
+        w2v_model = None
 
     train_loss = 0
 
     batch_size = int(args.batch_size)
-    num_files = len(fn)
+    num_files = zdh.num_files
     print("Number of files: ", num_files)
     for idx in range(0, num_files, batch_size):
         st = time.time()
 
-        batch = generate_batch(zip_file, fn, idx, batch_size, args.pass_string)
+        batch = zdh.generate_batch(idx, batch_size)
```

Comment:
What is the relationship between byte_embeddings.py and embed_binary.py? Comment suggests that one is done over bytecode, but this is not strictly accurate (we're training over raw bytes). One also sorts files by size before generating batches, which suggests that embed_binary represents the model as most recently discussed, and byte_embeddings is an older file that is retained in the source tree for historical reasons. Is this correct, or am I missing something key to why one gets size sorted and the other does not?